controllers/pod_models.go

```diff
@@ -280,18 +301,21 @@ func configureSidecarContainerForCluster(cluster *fdbtypes.FoundationDBCluster,
 		versionString = cluster.Spec.Version
 	}
 
-	return configureSidecarContainer(container, initMode, instanceID, versionString, cluster.Spec.SidecarVariables, cluster.Spec.FaultDomain, cluster.Spec.SidecarContainer, cluster.Spec.SidecarVersions, len(cluster.Spec.TrustedCAs) > 0)
+	publicIPSource := cluster.Spec.Services.PublicIPSource
+	usePublicIPFromService := publicIPSource != nil && *publicIPSource == fdbtypes.PublicIPSourceService
```

Comment:
This could be simplified to `usePublicIPFromService := cluster.NeedsExplicitListenAddress()`

Reply:
or you set it directly as argument without assigning a separate variable.

Reply:
In my head, I'm distinguishing between those two things, since we likely want to move to explicit listen addresses everywhere in the future. It's a little complicated in how it's being used, though, so I'll need to take a look at that. The way we're passing information to the configureSidecar method is getting unsustainable.